Add command to change the tld of local proxy (issue #159)

Nihilus118 opened the pull request:
PR for issue #159

tucksaun commented on commands/local_proxy_domain_attach.go:
```diff
@@ -48,7 +51,16 @@ var localProxyAttachDomainCmd = &console.Command{
 			return err
 		}
-		if err := config.AddDirDomains(projectDir, c.Args().Tail()); err != nil {
+
+		var domains []string
+		if strings.HasPrefix(c.Args().Slice()[0], ".") {
+			config.TLD = strings.TrimPrefix(c.Args().Slice()[0], ".")
+			domains = c.Args().Slice()[1:]
+		} else {
+			domains = c.Args().Slice()
+		}
+
+		if err := config.AddDirDomains(projectDir, domains); err != nil {
```
I don't think this makes sense to allow attaching and changing the TLD at the same time: this cover a quite small use case and it could be confusing for the end-user letting them think this TLD is only for the specific project (which is not the case).

Nihilus118 replied:
You are right. I reverted the file back to it's previous state.

tucksaun reviewed (commented):
Thank you for this contribution.

Here are a couple of comments to move forward.

Nihilus118 commented on commands/local_proxy_change_tld.go:
```diff
@@ -0,0 +1,55 @@
+	Name:     "proxy:change:tld",
+	Aliases:  []*console.Alias{{Name: "proxy:change:tld"}},
+	Usage:    "Change the TLD for the proxy",
+	Flags: []console.Flag{
+		dirFlag,
+	},
```
Removed.

Nihilus118 commented on commands/local_proxy_change_tld.go:
```diff
@@ -0,0 +1,55 @@
+			return err
+		}
+
+		config.TLD = c.Args().Get("tld")
```
What do you think about the regular expression `^[a-z0-9-]{2,63}$` for validating the tld before saving?

tucksaun replied:
IIRC, the top level domain is only allowed by the RFC to be composed of 1 to 63 alpha chars (except for IDN but I'm not sure this makes sense to support them for now)

Nihilus118 replied:
I adjusted the regular expression accordingly.

fabpot commented on commands/local_proxy_change_tld.go:
```diff
@@ -0,0 +1,58 @@
+
+var localProxyTLD = &console.Command{
+	Category: "local",
+	Name:     "proxy:change:tld",
```
```suggestion
	Name:     "proxy:tld",
```

With no arguments, it would return the current TLD, and set it if an argument is provided.

fabpot commented on commands/local_proxy_change_tld.go:
```diff
@@ -0,0 +1,58 @@
+	Category: "local",
+	Name:     "proxy:change:tld",
+	Aliases:  []*console.Alias{{Name: "proxy:change:tld"}},
+	Usage:    "Change the TLD for the proxy",
```
```suggestion
	Usage:    "Display or change the TLD for the proxy",
```

fabpot commented on commands/local_proxy_change_tld.go:
```diff
@@ -0,0 +1,58 @@
+			return err
+		}
+
+		terminal.Printfln("<info>The proxy is now configured with the following tld: %s</>", config.TLD)
```
```suggestion
		terminal.Printfln("<info>The proxy is now configured with the following TLD: %s</>", config.TLD)
```

fabpot reviewed (changes requested):
LGTM, I've just made a small suggestion, but maybe I'm overthinking it.

Nihilus118 commented:
> LGTM, I've just made a small suggestion, but maybe I'm overthinking it.

You are right, this is a better description of the command. I adjusted the filename for consistent naming, added another alias, made the argument `tld` optional and wrapped the logic for adjusting the TLD in a call to `c.Args().Present()` to prevent this error which occured when no new TLD was provided:
![image](https://github.com/symfony-cli/symfony-cli/assets/60194885/ce914360-b19b-41e3-9bb0-85811e0fcd1b)

fabpot reviewed (approved):
Thank you @Nihilus118
